src/serac/physics/utilities/finite_element_state.cpp

```diff
@@ -61,6 +65,28 @@ FiniteElementState& FiniteElementState::operator=(const double value)
   return *this;
 }
 
+double avg(const FiniteElementState& state)
+{
+  auto [count, _] = getMPIInfo(state.comm());
+  double sum, local_avg = state.trueVec().Sum() / state.trueVec().Size();
+  MPI_Allreduce(&local_avg, &sum, 1, MPI_DOUBLE, MPI_SUM, state.comm());
+  return sum / count;
+}
```

Comment:
Taking the average of the individual processor averages doesn't give the global average, in general. For example the vector {1, 2, 3, 4, 5} has a mean of 3, but if it is partitioned across two processors {1, 2} and {3, 4, 5}, averaging means of those two processors (1.5 and 4, respectively) gives 2.75 rather than 3.

I think the code below gets the averaging right (conceptually), but I'm not very proficient with MPI, so someone with more experience might want to double check these calls.
```cpp
double global_sum, local_sum = state.trueVec().Sum();
int global_size, local_size = state.trueVec().Size();
MPI_Allreduce(&local_sum, &global_sum, 1, MPI_DOUBLE, MPI_SUM, state.comm());
MPI_Allreduce(&local_size, &global_size, 1, MPI_INT, MPI_SUM, state.comm());
return global_sum / global_size;
```

Reply:
This is a good point. I will change it.  Also global/local is more clear I will update all the functions.